src/geanyobject.c

```diff
@@ -380,3 +389,13 @@ GObject *geany_object_new(void)
 {
 	return g_object_new(GEANY_OBJECT_TYPE, NULL);
 }
+
+/** Get the global GeanyObject instance
+ *
+ * @return @transfer{none} The instance
+ */
+GEANY_API_SYMBOL
+GObject *geany_object_get_instance(void)
```

Comment:
Okay, that part explains why you can't use a custom function.  Though, note that this means that pygobject needs to manage the signal connection lifetime itself well enough not to cause the issues our `plugin_signal_connect()` addresses (i.e. disconnect every signal handlers it still has registered when the plugin is unloaded).